de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/externallyreferenced/ExternallyReferencedCandidateRepository.java

```diff
@@ -288,4 +292,34 @@ private ExternallyReferencedCandidate forRecord(@NonNull final I_C_Invoice_Candi
 
 		return candidate.build();
 	}
+
+	/**
+	 * Persists the given invoice detail items that are not considered empty.
+	 * @see InvoiceDetailItem#isEmpty()
+	 *
+	 * @param invoiceCandidateId	Id of the invoice candidate to which the detail items are referring.
+	 * @param invoiceDetailItems    TO holding additional details about invoice candidate.
+	 */
+	public void storeInvoiceDetailItems(final InvoiceCandidateId invoiceCandidateId, final List<InvoiceDetailItem> invoiceDetailItems)
```

Comment:
As pritten further up; Eric Evans would love to see this this method be not public, but called from `save(..)`.
(feel free to contact @teosarca or me for details)